db/users.go

```diff
@@ -128,6 +129,21 @@ func (c *userCollection) DeleteUser(ctx context.Context, userID string) error {
 	return nil
 }
 
+func (c *userCollection) DeleteUsersBySessionID(ctx context.Context, sessionID string) error {
+	errMsg := "[db] delete users by session id: %w"
+
+	filter := bson.M{"session_id": sessionID}
+	res, err := c.collection.DeleteMany(ctx, filter)
+	if err != nil {
+		return fmt.Errorf(errMsg, err)
+	}
+	if res.DeletedCount == 0 {
+		return fmt.Errorf(errMsg, ErrNoSessionWithID)
+	}
```

Comment:
i think it makes more sense to check for matchedCount

Reply:
The result of a DeleteMany operation only supplies a DeletedCount.